cmd/kubeadm/app/cmd/join.go

```diff
@@ -225,60 +225,48 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 	return cmd
 }
 
-// AddJoinConfigFlags adds join flags bound to the config to the specified flagset
-func AddJoinConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1beta1.JoinConfiguration) {
+// addJoinConfigFlags adds join flags bound to the config to the specified flagset
+func addJoinConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1beta1.JoinConfiguration) {
 	flagSet.StringVar(
 		&cfg.NodeRegistration.Name, options.NodeName, cfg.NodeRegistration.Name,
 		`Specify the node name.`,
 	)
+	// adds file control plane flags to the specified flagset
 	flagSet.StringVar(
-		&cfg.Discovery.TLSBootstrapToken, options.TLSBootstrapToken, cfg.Discovery.TLSBootstrapToken,
-		`Specify the token used to temporarily authenticate with the Kubernetes Master while joining the node.`,
+		&cfg.ControlPlane.LocalAPIEndpoint.AdvertiseAddress, options.APIServerAdvertiseAddress, cfg.ControlPlane.LocalAPIEndpoint.AdvertiseAddress,
+		"If the node should host a new control plane instance, the IP address the API Server will advertise it's listening on. If not set the default network interface will be used.",
 	)
-	AddControlPlaneFlags(flagSet, cfg.ControlPlane)
-	AddJoinBootstrapTokenDiscoveryFlags(flagSet, cfg.Discovery.BootstrapToken)
-	AddJoinFileDiscoveryFlags(flagSet, cfg.Discovery.File)
-	cmdutil.AddCRISocketFlag(flagSet, &cfg.NodeRegistration.CRISocket)
-}
-
-// AddJoinBootstrapTokenDiscoveryFlags adds bootstrap token specific discovery flags to the specified flagset
-func AddJoinBootstrapTokenDiscoveryFlags(flagSet *flag.FlagSet, btd *kubeadmapiv1beta1.BootstrapTokenDiscovery) {
+	flagSet.Int32Var(
+		&cfg.ControlPlane.LocalAPIEndpoint.BindPort, options.APIServerBindPort, cfg.ControlPlane.LocalAPIEndpoint.BindPort,
+		"If the node should host a new control plane instance, the port for the API Server to bind to.",
+	)
+	// adds bootstrap token specific discovery flags to the specified flagset
 	flagSet.StringVar(
-		&btd.Token, options.TokenDiscovery, "",
+		&cfg.Discovery.BootstrapToken.Token, options.TokenDiscovery, "",
 		"For token-based discovery, the token used to validate cluster information fetched from the API server.",
 	)
 	flagSet.StringSliceVar(
-		&btd.CACertHashes, options.TokenDiscoveryCAHash, []string{},
+		&cfg.Discovery.BootstrapToken.CACertHashes, options.TokenDiscoveryCAHash, []string{},
 		"For token-based discovery, validate that the root CA public key matches this hash (format: \"<type>:<value>\").",
 	)
 	flagSet.BoolVar(
-		&btd.UnsafeSkipCAVerification, options.TokenDiscoverySkipCAHash, false,
+		&cfg.Discovery.BootstrapToken.UnsafeSkipCAVerification, options.TokenDiscoverySkipCAHash, false,
 		"For token-based discovery, allow joining without --discovery-token-ca-cert-hash pinning.",
 	)
-}
-
-// AddJoinFileDiscoveryFlags adds file discovery flags to the specified flagset
-func AddJoinFileDiscoveryFlags(flagSet *flag.FlagSet, fd *kubeadmapiv1beta1.FileDiscovery) {
+	//	discovery via kube config file flag
```

Comment:
Please, convert the tab into a space here.

Reply:
i think this is a github display issue, i checked it is a space

Reply:
looks like a tab in the email DIFF too.

Reply:
Checked out this version of the file and opened it in vim, it's a tab. Please, fix that in order to keep the code well aligned and I'll lgtm it.

Reply:
done @rosti thx :rocket: